pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -353,12 +354,30 @@ def on_save_checkpoint(
     def on_load_checkpoint(
         self, trainer: "pl.Trainer", pl_module: "pl.LightningModule", callback_state: Dict[str, Any]
     ) -> None:
-        self.best_model_score = callback_state["best_model_score"]
-        self.best_model_path = callback_state["best_model_path"]
-        self.best_k_models = callback_state.get("best_k_models", self.best_k_models)
-        self.kth_best_model_path = callback_state.get("kth_best_model_path", self.kth_best_model_path)
-        self.kth_value = callback_state.get("kth_value", self.kth_value)
-        self.last_model_path = callback_state.get("last_model_path", self.last_model_path)
+        # Need to get real path from the given `dirpath`, we follow the template from
+        # __init_ckpt_dir function to get the real path for comparison
+        dirpath_from_ckpt = callback_state.get("dirpath", self.dirpath)
+        _fs = get_filesystem(dirpath_from_ckpt if dirpath_from_ckpt else "")
+
+        if dirpath_from_ckpt and _fs.protocol == "file":
+            dirpath_from_ckpt = os.path.realpath(dirpath_from_ckpt)
+
+        # For more context, see: https://github.com/PyTorchLightning/pytorch-lightning/issues/11379
```

Comment:
You don't need to link to the issue or PR, git blame can be used to find the context

Reply:
Sure, removed now.